src/Templates/BaseDate.java

```diff
@@ -0,0 +1,86 @@
+package Templates;
+
+import java.io.Serializable;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.*;
+
+import Exceptions.MarioDateTimePassed;
+
+public class BaseDate implements Serializable {
+    private static final List<String> dateFormats = List.of(
+            "yyyy-MM-dd",
+            "dd-MM-yyyy",
+            "dd/MM/yyyy",
+            "yyyy-MM-dd",
+            "yyyy/MM/dd",
+            "dd:MM:yyyy",
+            "ddMMyyyy",
+            "ddMMyy");
+
+    private static final List<String> timeFormats = List.of(
+            "HH:mm",
+            "HHmm");
+
+    private static final ArrayList<String> dateTimeFormats = dateTimeVary();
+    public static DateTimeFormatter formatter = null;
+    LocalDateTime dateTime = null;
+
+    public BaseDate(String args) throws Exception {
+        args = args.strip();
+        if (!args.contains(" ")) {
+            args = args + " 0000";
```

Comment:
Maybe using a named constant for " 0000" would be more appropriate to avoid magic literals?